StRoot/StarGenerator/Hijing1_383/StarHijing.cxx

```diff
@@ -98,6 +119,17 @@ Int_t StarHijing::LuComp( Int_t jetsetid )
 Int_t StarHijing::Init()
 {
 
+  // Configuration based on StMaker attributes
+  if ( SAttr("FRAME") ) SetFrame( SAttr("FRAME"), DAttr("Ecms") );
+  if ( SAttr("BLUE")  ) SetBlue ( SAttr("BLUE" ) );
+  if ( SAttr("YELL")  ) SetYell ( SAttr("YELL" ) );
+  if ( SAttr("SET" )  ) {
+    for ( auto cmd : tokenize( SAttr("Set"), ";" ) ) {
+      Set( cmd.c_str() );
+    }
+  }
```

Comment:
Pros and cons of placing this in a new StarGenerator::Init() and having classes which inherit from StarGenerator call StarGenerator::Init() if they have an Init()? I think all of this could be placed there, in which case you only need one tokenize() as well, located in StarGenerator.cxx.